BUG: `numpy.core.tests.test_mem_policy.test_new_policy` fails on musllinux

### Describe the issue:

A CI run is being created for musllinux in https://github.com/numpy/numpy/pull/22864. This has uncovered several test fails. One of these is a test `numpy.core.tests.test_mem_policy.test_new_policy`

### Reproduce the code example:

```python
NA
```


### Error message:

```shell
FAILED numpy/core/tests/test_mem_policy.py::test_new_policy - AssertionError:...

2022-12-22T01:17:48.0273995Z _______________________________ test_new_policy ________________________________
2022-12-22T01:17:48.0274173Z 
2022-12-22T01:17:48.0275059Z get_module = <module 'mem_policy' from '/tmp/pytest-of-root/pytest-0/test_set_policy0/mem_policy/mem_policy.cpython-310-x86_64-linux-gnu.so'>
2022-12-22T01:17:48.0275518Z 
2022-12-22T01:17:48.0275618Z     @pytest.mark.slow
2022-12-22T01:17:48.0275867Z     def test_new_policy(get_module):
2022-12-22T01:17:48.0276099Z         a = np.arange(10)
2022-12-22T01:17:48.0276399Z         orig_policy_name = np.core.multiarray.get_handler_name(a)
2022-12-22T01:17:48.0276992Z     
2022-12-22T01:17:48.0277248Z         orig_policy = get_module.set_secret_data_policy()
2022-12-22T01:17:48.0277487Z     
2022-12-22T01:17:48.0277808Z         b = np.arange(10)
2022-12-22T01:17:48.0278252Z         assert np.core.multiarray.get_handler_name(b) == 'secret_data_allocator'
2022-12-22T01:17:48.0278541Z     
2022-12-22T01:17:48.0278787Z         # test array manipulation. This is slow
2022-12-22T01:17:48.0279171Z         if orig_policy_name == 'default_allocator':
2022-12-22T01:17:48.0279500Z             # when the np.core.test tests recurse into this test, the
2022-12-22T01:17:48.0279994Z             # policy will be set so this "if" will be false, preventing
2022-12-22T01:17:48.0280299Z             # infinite recursion
2022-12-22T01:17:48.0280511Z             #
2022-12-22T01:17:48.0280741Z             # if needed, debug this by
2022-12-22T01:17:48.0281152Z             # - running tests with -- -s (to not capture stdout/stderr
2022-12-22T01:17:48.0281538Z             # - setting extra_argv=['-vv'] here
2022-12-22T01:17:48.0281961Z >           assert np.core.test('full', verbose=2, extra_argv=['-vv'])
2022-12-22T01:17:48.0282294Z E           AssertionError: assert False
2022-12-22T01:17:48.0282869Z E            +  where False = <numpy._pytesttester.PytestTester object at 0x7f2722d31d50>('full', verbose=2, extra_argv=['-vv'])
2022-12-22T01:17:48.0283963Z E            +    where <numpy._pytesttester.PytestTester object at 0x7f2722d31d50> = <module 'numpy.core' from '/__w/numpy/numpy/build/testenv/lib/python3.10/site-packages/numpy/core/__init__.py'>.test
2022-12-22T01:17:48.0284718Z E            +      where <module 'numpy.core' from '/__w/numpy/numpy/build/testenv/lib/python3.10/site-packages/numpy/core/__init__.py'> = np.core
2022-12-22T01:17:48.0285156Z 
2022-12-22T01:17:48.0285279Z a          = array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
2022-12-22T01:17:48.0285533Z b          = array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
2022-12-22T01:17:48.0286103Z get_module = <module 'mem_policy' from '/tmp/pytest-of-root/pytest-0/test_set_policy0/mem_policy/mem_policy.cpython-310-x86_64-linux-gnu.so'>
2022-12-22T01:17:48.0286570Z orig_policy = <capsule object "mem_handler" at 0x7f271e8d5c80>
2022-12-22T01:17:48.0286933Z orig_policy_name = 'default_allocator'


```


### Runtime information:

python3.10, numpy main, musllinux_x86_64

### Context for the issue:

_No response_

--- a/numpy/core/tests/test_mem_policy.py
+++ b/numpy/core/tests/test_mem_policy.py
@@ -5,7 +5,7 @@ import pytest
 import numpy as np
 import threading
 import warnings
-from numpy.testing import extbuild, assert_warns, IS_WASM, IS_MUSL
+from numpy.testing import extbuild, assert_warns, IS_WASM
 import sys
 
 
@@ -358,7 +358,6 @@ def test_thread_locality(get_module):
     assert np.core.multiarray.get_handler_name() == orig_policy_name
 
 
-@pytest.mark.xfail(IS_MUSL, reason="gh23050")
 @pytest.mark.slow
 def test_new_policy(get_module):
     a = np.arange(10)

